packages/source-iotsitewise/src/time-series-data/client/batch.ts

```diff
@@ -48,11 +49,19 @@ export function createRawLatestEntryBatches<T extends { maxResults?: number }>(
 }
 
 export function createRawHistoricalEntryBatches<
-  T extends { maxResults?: number }
+  T extends { requestInformation?: RequestInformation; maxResults?: number }
 >(entries: T[]): [T[], number][] {
   const buckets: Record<number, T[]> = {};
+  const singles: [T[], number][] = [];
```

Comment:
Nit: It would be beneficial to have a comment here (or somewhere) explaining why this is done (i.e., prevent grouping of singles).